src/macaron/dependency_analyzer/repo_finder.py

```diff
@@ -0,0 +1,185 @@
+
+from macaron.config.global_config import global_config
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+def create_urls(group: str, artifact: str, version: str) -> list[str]:
+    """
+    Create the urls to search for the pom relating to the passed GAV.
+
+    Parameters
+    ----------
+    group : str
+        The group ID.
+    artifact: str
+        The artifact ID.
+    version: str
+        The version of the artifact.
+
+    Returns
+    -------
+    list[str]
+        The list of created URLs.
+    """
+    urls = []
+
+    version_pruned = ""
+    if "-" in version:
+        # Create a pruned version to fix some strangely labeled artifacts
+        version_pruned = re.sub("-[a-zA-Z]+", "", version)
+
+    # Create URLs using all configured repositories
+    for repo in global_config.artifact_repositories:
+        urls.append(f"{repo}/{group}/{artifact}/{version}/{artifact}-{version}.pom")
+        if version_pruned != "":
+            urls.append(f"{repo}/{group}/{artifact}/{version_pruned}/{artifact}-{version_pruned}.pom")
+
+    return urls
+
+
+def parse_gav(gav: str) -> list[str]:
+    """
+    Parse the passed GAV string, splitting it into its constituent parts.
+
+    Parameters
+    ----------
+    gav : str
+        An artifact represented as a GAV (group, artifact, version) in the format: G:A:V.
+
+    Returns
+    -------
+    list[str]
+        The list of URLs created for the GAV by the create_urls trailing function call.
+    """
+    split = gav.split(":")
+    if len(split) < 3:
+        logger.error("Could not parse GAV: %s", gav)
+        return []
+
+    group = split[0]
+    artifact = split[1]
+    version = split[2]
+    group_url = group.replace(".", "/")
+
+    return create_urls(group_url, artifact, version)
+
+
+def retrieve_pom(session: requests.Session, url: str) -> str:
+    """
+    Attempt to retrieve the file located at the passed URL using the passed Session.
+
+    Parameters
+    ----------
+    session : requests.Session
+        The HTTP session to use for attempting the GET request.
+    url : str
+        The URL for the GET request.
+
+    Returns
+    -------
+    str :
+        The retrieved file data or an empty string.
+    """
+    if not url.endswith(".pom"):
+        return ""
+    res = session.get(url)
+    if not res.ok:
+        logger.warning("Failed to retrieve pom from: %s, error code: %s", url, res.status_code)
+        return ""
+    logger.info("Found artifact POM at: %s", url)
+    return res.text
+
+
+def _find_element(parent: typing.Any, target: str) -> typing.Any:
+    # Attempt to match the target tag within the children of parent.
+    for child in parent:
+        # Account for raw tags and tags accompanied by Maven metadata enclosed in curly braces. E.g. '{metadata}tag'
+        if child.tag == target or child.tag.endswith("}" + target):
+            return child
+    return None
+
+
+def parse_pom(pom: str, tags: list[str]) -> list[str]:
```

Comment:
Can you please use an iterator instead of creating a list as the return value to improve the performance?